### ISSUE TITLE
utils/Utils.cpp depending on http/HttpClient_Android.hpp?

### ISSUE THREAD

**Issue by maxgolov** [microsoft/cpp_client_telemetry#548]
@larvacea - Martin, quick question - let me know your thoughts on this..

I've been working on a hackathon project (integrating OpenTelemetry C++ core instead of our 1DS core), and I noticed one interesting detail. It seems like **only for Android** - while figuring out what temp dir to use for DB files and stuff, we invoke something like this:

https://github.com/microsoft/cpp_client_telemetry/blob/b228d1b82522cf131a41f037938448336ed855d3/lib/utils/Utils.cpp#L108

to obtain temporary dir (app cache) path. That way we actually create tight coupling between a general routine to obtain TEMP dir path and Android HTTP client implementation, that could otherwise be avoided.

I wonder if we could have something like this instead:
https://stackoverflow.com/questions/7595324/creating-temporary-files-in-android-with-ndk
Basically a set of Android-specific helper Utils (maybe under **lib/android_build** or as **lib/utils/android/Utils.cpp**) that allow to decouple the routine to obtain temp (cache dir) path from HTTP client code? This would also help eventually when Edge-on-Android binds to their own Chromium stack (as their own implementation may not necessary need to implement the `GetCacheFilePath` in their HTTP client at all).
